core/iwasm/compilation/aot_emit_memory.c

```diff
@@ -202,9 +203,32 @@ aot_check_memory_overflow(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
                     goto fail;
                 }
             }
+            if (alignp != NULL) {
+                /*
+                 * A note about max_align below:
+                 * the assumption here is the base address of a linear memory
+                 * has the natural alignment. for platforms using mmap, it can
+                 * be even larger. for now, use a conservative value.
+                 */
+                const int max_align = 4;
```

Comment:
Currently the linear memory is allocated with os_mmap and normally the returned address is 8-byte aligned, some platforms have applied special handling to ensure that is 8-byte aligned, e.g. esp-idf and nuttx:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/shared/platform/esp-idf/espidf_memmap.c#L39-L41

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/shared/platform/nuttx/nuttx_platform.c#L82

So is it better to set max_align as 8 here? In my memory, 8-byte aligned is an implicit requirement for memory allocation in WAMR, but we can make it explicit and add assert for os_mmap in each platform implementation?

Reply:
ok.